pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -208,6 +209,37 @@ func (n *nodeHealthMap) set(name string, data *nodeHealthData) {
 	n.nodeHealths[name] = data
 }
 
+// retryMarkPodsNotReadySet stores names of nodes that require
```

Comment:
What you basically do here is creating thread-safe set.
So instead of creating new structures, let's use: https://golang.org/pkg/sync/#Map
[basically map from string -> struct{}{}]

Reply:
Done.